### PR TITLE
Java: dbscheme and library changes related to Java 14 extractor upgrade

### PR DESCRIPTION
**Pull request by yo-h**
Depends on corresponding internal PR for the Java 14 extractor upgrade.

Standard language features in Java 14 will be supported. Full CodeQL-level support for preview features will follow in a subsequent PR.

Tests are expected to fail on this PR, but should succeed on the internal PR.

### CODE REVIEW

File: java/ql/test/library-tests/structure/DeclaresMember.expected [hunk 1]
```diff
@@ -28,7 +28,6 @@
 | LocalClass | LocalClass |
 | LocalClass | n |
 | MemberClass | MemberClass |
-| Object | <clinit> |
```
[the remark this answers is not in the document]
Comment (yo-h) [reply java/ql/test/library-tests/structure/DeclaresMember.expected:31]: This is expected. The static initializer of `Object` was [removed](https://hg.openjdk.java.net/jdk-updates/jdk14u/comparison/ba524a5f7cc2/src/java.base/share/classes/java/lang/Object.java) in Java 14.

File: java/ql/test/library-tests/structure/EnclosingCallables.expected [hunk 2]
```diff
@@ -1,12 +1,19 @@
+| structure/A.java:5:14:5:14 | void | A |
```
[the remark this answers is not in the document]
Comment (yo-h) [reply java/ql/test/library-tests/structure/EnclosingCallables.expected:1]: The addition of a `void` type access for synthetic default constructors was a side-effect of a change in our frontend. I'll add a commit to the internal extractor PR to suppress this.

**Comment by yo-h** [comment]
Rebased to resolve submodule conflict on internal PR.

File: java/ql/test/library-tests/structure/EnclosingCallables.expected (hunk 2 above)
[the remark this answers is not in the document]
Comment (aibaars) [reply java/ql/test/library-tests/structure/EnclosingCallables.expected]: Great, thanks!

File: java/ql/test/library-tests/structure/DeclaresMember.expected (hunk 1 above)
[the remark this answers is not in the document]
Comment (aibaars) [reply java/ql/test/library-tests/structure/DeclaresMember.expected:31]: Ah ok, that make sense.

**Pull request merged**